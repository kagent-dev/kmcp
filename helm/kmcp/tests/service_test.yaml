suite: Test service template
templates:
  - service.yaml

tests:
  - it: should create service when metrics enabled
    template: service.yaml
    set:
      controller.metrics.enabled: true
      service.type: ClusterIP
      service.port: 8443
      service.targetPort: 8443
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should not create service when metrics disabled
    template: service.yaml
    set:
      controller.metrics.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should set custom service type
    template: service.yaml
    set:
      controller.metrics.enabled: true
      service.type: LoadBalancer
      service.port: 8443
      service.targetPort: 8443
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set custom service port
    template: service.yaml
    set:
      controller.metrics.enabled: true
      service.type: ClusterIP
      service.port: 9090
      service.targetPort: 8443
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set custom target port
    template: service.yaml
    set:
      controller.metrics.enabled: true
      service.type: ClusterIP
      service.port: 8443
      service.targetPort: 9090
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {} 