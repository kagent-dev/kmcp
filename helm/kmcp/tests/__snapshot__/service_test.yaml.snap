should create service when metrics enabled:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager-metrics-service
      namespace: NAMESPACE
    spec:
      ports:
        - name: https
          port: 8443
          protocol: TCP
          targetPort: 8443
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
      type: ClusterIP
should set custom service port:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager-metrics-service
      namespace: NAMESPACE
    spec:
      ports:
        - name: https
          port: 9090
          protocol: TCP
          targetPort: 8443
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
      type: ClusterIP
should set custom service type:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager-metrics-service
      namespace: NAMESPACE
    spec:
      ports:
        - name: https
          port: 8443
          protocol: TCP
          targetPort: 8443
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
      type: LoadBalancer
should set custom target port:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager-metrics-service
      namespace: NAMESPACE
    spec:
      ports:
        - name: https
          port: 8443
          protocol: TCP
          targetPort: 9090
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
      type: ClusterIP
